### PR TITLE
feat(kit): support `fallback` for `avatar` when img not found by `avatarUrl`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

image

### CODE REVIEW

File: projects/kit/components/avatar/avatar.component.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export class TuiAvatarComponent {
     @tuiDefaultProp()
     text = '';
 
+    @Input()
+    @tuiDefaultProp()
+    fallback: SafeHtml | string | null = null;
```
Comment (nsbarsukov) [inline projects/kit/components/avatar/avatar.component.ts:40]: Add new input's description to https://taiga-ui.dev/components/avatar/API

File: projects/demo/src/modules/components/avatar/examples/6/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    styleUrls: ['./index.less'],
+    changeDetection,
+    encapsulation,
+})
+export class TuiAvatarExample6 {
+    readonly waterplea = 'https://avatars.githubusercontent.com/u/11832552?v=4';
+    readonly wrongUrl = 'https://google.com/images/not-found.png';
+
+    readonly fallback = this.sanitizer.bypassSecurityTrustHtml(
+        '<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em"><g xmlns="http://www.w3.org/2000/svg"><svg x="50%" y="50%" width="1em" height="1em" overflow="visible" viewBox="0 0 16 16"><svg x="-8" y="-8"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"><path d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"/><circle cx="12" cy="7" r="4"/></svg></svg></svg></g></svg>',
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/avatar/examples/6/index.ts:18]: Should we simplify this example ? 

```html
<tui-avatar
    class="avatar"
    [avatarUrl]="waterplea"
></tui-avatar>

<tui-avatar
    class="tui-space_left-3 avatar"
    fallback="tuiIconUserLarge"
    [avatarUrl]="wrongUrl"
></tui-avatar>
```

Or does everything exist on purpose ?

File: projects/demo/src/modules/components/avatar/examples/6/index.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    templateUrl: './index.html',
+    styleUrls: ['./index.less'],
+    changeDetection,
+    encapsulation,
+})
+export class TuiAvatarExample6 {
+    readonly waterplea = 'https://avatars.githubusercontent.com/u/11832552?v=4';
+    readonly wrongUrl = 'https://google.com/images/not-found.png';
+
+    readonly fallback = this.sanitizer.bypassSecurityTrustHtml(
```
Comment (waterplea) [inline projects/demo/src/modules/components/avatar/examples/6/index.ts:17]: Avatar works with tuiIcon icons, let's not add this and just use a named icon as a fallback.

File: projects/kit/components/avatar/avatar.component.ts [hunk 4]
```diff
@@ -65,22 +71,26 @@ export class TuiAvatarComponent {
...
+            : this.getSlicedText(this.text, this.size);
     }
 
     get stringAvatar(): string {
         return this.iconAvatar ? String(this.avatarUrl) : '';
     }
 
     onError(): void {
+        this.hasError = true;
         this.isUrlValid = false;
```
Comment (waterplea) [inline projects/kit/components/avatar/avatar.component.ts:85]: What `isUrlValid` is used for? Can we reuse it without introducing another flag?

File: projects/kit/components/avatar/avatar.component.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/avatar/avatar.component.ts:40]: fixed

File: projects/demo/src/modules/components/avatar/examples/6/index.ts (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/components/avatar/examples/6/index.ts]: fixed

File: projects/kit/components/avatar/avatar.component.ts (hunk 4 above)
Comment (splincode) [reply projects/kit/components/avatar/avatar.component.ts:88]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**